bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -710,35 +727,70 @@ private Runnable getStorer(final byte[] data, final int readingOffset, final int
         @Override
         public void run()
         {
-          int i = storage.store(identifier, data, readingOffset, writingOffset);
-          if (i == 0) {
+          if (uniqueIdentifier == 0) {
```

Comment:
Agree, but this needs to be defined by Storage interface, not by DataList or DataList.Block. Please file JIRA.